examples/SampleUtils.h

```diff
@@ -120,45 +120,16 @@ namespace utils {
         const ml::Operand& operand;
     } NamedOperand;
 
-    ml::Graph AwaitBuild(const ml::GraphBuilder& builder, const std::vector<NamedOperand>& outputs);
+    ml::Graph Build(const ml::GraphBuilder& builder, const std::vector<NamedOperand>& outputs);
 
     typedef struct {
         const std::string name;
-        const ml::Input input;
-    } NamedInput;
+        const std::vector<float>& resource;
+    } NamedResource;
 
-    typedef struct {
-        const std::string name;
-        const ml::Output output;
-    } NamedOutput;
-
-    ml::NamedResults AwaitCompute(const ml::Graph& compilation,
-                                  const std::vector<NamedInput>& inputs,
-                                  const std::vector<NamedOutput>& outputs = {});
-
-    bool CheckShape(const ml::Result& result, const std::vector<int32_t>& expectedShape);
-
-    template <class T>
-    bool CheckValue(const ml::Result& result, const std::vector<T>& expectedValue) {
-        if (result.GetHandle() == nullptr) {
-            return false;
-        }
-        size_t size = result.BufferSize() / sizeof(T);
-        if (size != expectedValue.size()) {
-            dawn::ErrorLog() << "The size of output data is expected as " << expectedValue.size()
-                             << ", but got " << size;
-            return false;
-        }
-        for (size_t i = 0; i < result.BufferSize() / sizeof(T); ++i) {
-            T value = static_cast<const T*>(result.Buffer())[i];
-            if (!Expected(value, expectedValue[i])) {
-                dawn::ErrorLog() << "The output value at index " << i << " is expected as "
-                                 << expectedValue[i] << ", but got " << value;
-                return false;
-            }
-        }
-        return true;
-    }
+    ml::ComputeGraphStatus Compute(const ml::Graph& graph,
+                                   const std::vector<NamedResource>& inputs,
+                                   const std::vector<NamedResource>& outputs);
```

Comment:
This would be a template function as well. For convenience, we can make a specialized version for float.